jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -1104,18 +1102,18 @@ public void clearWarnings() throws SQLException {
 
   @Override
   public void close() throws SQLException {
-    if (!isClosed) {
-      TCloseSessionReq closeReq = new TCloseSessionReq(sessHandle);
       try {
-        client.CloseSession(closeReq);
+        if (!isClosed) {
+          TCloseSessionReq closeReq = new TCloseSessionReq(sessHandle);
+          client.CloseSession(closeReq);
+        }
       } catch (TException e) {
         throw new SQLException("Error while cleaning up the server resources", e);
       } finally {
         isClosed = true;
-        if (transport != null) {
+        if (transport != null && transport.isOpen()) {
           transport.close();
         }
```

Comment:
can we release the references as well?
```
transport=null
client=null
```

Reply:
done and add some pre-checks on the connection state to avoid potential NPE